### PR TITLE
Remove unneeded package dependencies

### PR DESCRIPTION
**Pull request by xugao**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

- Remove unneeded package dependencies
- Add missing set-version for packages.
- Enable ESLint for `date-time-utilites`

#### Focus areas to test

(optional)

### CODE REVIEW

File: apps/codesandbox-react-northstar-template/package.json [hunk 1]
```diff
@@ -18,9 +18,7 @@
     "typescript": "3.7.2"
   },
   "dependencies": {
-    "@fluentui/accessibility": "^0.51.0",
```
Comment (ecraig12345) [inline apps/codesandbox-react-northstar-template/package.json:21]: These ones might be needed to get it to substitute the PR versions?

File: packages/date-time-utilities/package.json [hunk 2]
```diff
@@ -24,7 +24,6 @@
     "update-api": "just-scripts update-api"
   },
   "devDependencies": {
-    "@fluentui/eslint-plugin": "^0.54.1",
```
Comment (ecraig12345) [inline packages/date-time-utilities/package.json:27]: This needs to stay too--the fact that this package doesn't have eslintrc is wrong.

File: packages/fluentui/perf/package.json [hunk 3]
```diff
@@ -5,13 +5,11 @@
   "license": "MIT",
   "dependencies": {
     "@babel/polyfill": "^7.10.4",
-    "@fluentui/docs": "^0.51.0",
```
Comment (ecraig12345) [inline packages/fluentui/perf/package.json:8]: This is used in this package.

File: packages/react-next/package.json [hunk 4]
```diff
@@ -35,7 +35,6 @@
   },
   "devDependencies": {
     "@fluentui/common-styles": "^0.2.6",
-    "@fluentui/eslint-plugin": "^0.54.1",
```
Comment (ecraig12345) [inline packages/react-next/package.json:38]: This one might need to stay even though it's not referenced directly (not sure)

File: packages/react-spinbutton/package.json [hunk 5]
```diff
@@ -26,7 +26,6 @@
   },
   "devDependencies": {
```
Comment (ecraig12345) [inline packages/react-spinbutton/package.json:27]: I thought I deleted this package? Or maybe that PR is still open.

**Comment by ecraig12345** [comment]
Build is failing while running storybook webpack in `date-time` (and probably other packages) with errors like `Module not found: Error: Can't resolve '@fluentui/react' in '/mnt/work/3/s/packages/react-examples/src/date-time/Calendar'`

I hit this in #15414 too, and I'm guessing the issue is that with the `example-app-base` dep removed, it also implicitly prevented `@fluentui/react` from getting added to the resolve alias. I added a console log of the generated aliases in that PR to help debug but haven't gotten to look back at it.

**Review by ecraig12345 — APPROVED** [review]
Thanks for doing this! Looks like it still needs a fix for the bundling issue though.

File: packages/date-time-utilities/src/dateGrid/getDayGrid.ts [hunk 6]
```diff
@@ -56,16 +56,14 @@ export const getDayGrid = (options: IDayGridOptions): IDay[][] => {
...
-    selectedDates = getDateRangeArray(
-      selectedDate,
-      selectedDateRangeType,
-      firstDayOfWeek,
-      workWeekDays,
-      daysToSelectInDayView,
-    );
-    selectedDates = getBoundedDateRange(selectedDates, minDate, maxDate);
-  }
+  selectedDates = getDateRangeArray(
```
Comment (ecraig12345) [inline packages/date-time-utilities/src/dateGrid/getDayGrid.ts:59]: Given the [northstar screener failures](https://screener.io/v2/states/H17FwCf88.microsoft-fluentui-fluentui/xgao%2Fremove-unused-deps/1024x768/Chrome/datepickercellexampleshorthandtsx-shows-datepicker-popup-0) I suspect the typings here are wrong (relative to usage), and the `if (selectedDate)` check is actually needed. Maybe best to add back the if() and update the typings to allow undefined?

Comment (xugao) [reply packages/date-time-utilities/src/dateGrid/getDayGrid.ts]: my bad, pushed fix!

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**